sklearn/tree/tree.py

```diff
@@ -269,6 +277,10 @@ def fit(self, X, y, sample_weight=None, check_input=True):
         min_samples_split = max(self.min_samples_split,
                                 2 * self.min_samples_leaf)
 
+        if X_idx_sorted is None and self.presort == True and not issparse(X):
+            X_idx_sorted = np.asfortranarray(np.argsort(X, axis=0),
+                                             dtype=np.int32)
```

Comment:
What is the benefit of sorting here instead of making this in the splitter?
If it's inside the splitter, it seems to remove a lot of complexity ovreall.